charms/worker/k8s/src/charm.py

```diff
@@ -98,24 +100,24 @@ def _check_k8sd_ready(self):
     )
     def _bootstrap_k8s_snap(self):
         """Bootstrap k8s if it's not already bootstrapped."""
-        # TODO: Remove `is_cluster_bootstrapped` check once
-        # https://github.com/canonical/k8s-snap/pull/99 landed.
+        if self.is_worker:
+            return
+
```

Comment:
there's nothing wrong with this -- but the `_reconcile(...)` method won't call this method.  

```python
        if self.unit.is_leader() and self.is_control_plane:
            self._bootstrap_k8s_snap()
            self._enable_components()
            self._create_cluster_tokens()
```